catroid/src/main/java/org/catrobat/catroid/utils/Utils.java

```diff
@@ -479,6 +476,15 @@ public static double round(double value, int precision) {
 		return (double) Math.round(value * scale) / scale;
 	}
 
+	private static void displayDialog(Context context, final Builder builder) {
+		((Activity) context).runOnUiThread(new Runnable() {
```

Comment:
I took another look at the changes. The quick-fix is currently the only solution. Please use `((Activity) context).runOnUiThread`.

The problem is that methods like `getCurrentProjectName`, `copyImageToCatroid` and so on interact with the ui. They definitely should *not* interact with the ui. Fixing it properly would need a looot of rewrite. So I'm fine with the quick-fix for now. Please revert to your initial version. Sorry!
No tests will be needed! :wink: